### DISCUSSION TITLE
Error When Launching Terminal

### DISCUSSION THREAD

**Discussion by Homer712** [Homebrew/discussions#5803] [Getting started] [answered]
### Output of `brew config`

```shell
Uninstalled/None
```

### Output of `brew doctor`

```shell
Uninstalled/None
```

### Description of issue

I uninstalled Homebrew, now when I launch Terminal (I'm on MacOS Sonoma 14.7.1) I get the following error:

Last login: Thu Dec 12 17:23:06 on ttys000
/Users/homer/.zprofile:2: no such file or directory: /opt/homebrew/bin/brew
homer@Homers-MacBook-Pro ~ %

**Comment by SMillerDev** [comment]
Did you also remove the brew setup from your `~/.zprofile` file?

**Comment by Homer712** [comment]
Thanks for the quick reply. I'm a "senior" and struggling here, so I have no idea if that file is removed during the uninstall (I used the provided script) process. I'm just trying to set the Terminal to use zsh.

**Reply by p-linnane** [reply]
You need to open the file at `~/.zprofile` and remove the line referencing Homebrew. It's probably `eval "$(/opt/homebrew/bin/brew shellenv)"`.

**Reply by Homer712** [reply]
You'll need to give me a hint here. Where would I find that file please?

**Reply by gromgit** [reply, marked as answer]
1. Run the following command to open all the relevant files in macOS' built-in editor: `open -a TextEdit ~/.zprofile ~/.zshrc`
2. Look for all lines that contain `/opt/homebrew/bin/brew` and delete them.
3. Save all the files.
4. Start a new Terminal window to verify that the fix works.

**Reply by p-linnane** [reply]
`~/.zprofile` is the full path of the file. `~` is your home directory (`/Users/<your-username>`). Since the file has a `.` at the beginning of the file name, it will not be visible within Finder unless you press Command + Shift + . at the same time.

**Comment by Homer712** [comment]
Thank you, found the file. I'll post below what the file's text is, please just tell me what it needs to look like after I edit it. And, thank you all for your patience.

# Set PATH, MANPATH, etc., for Homebrew.
eval "$(/opt/homebrew/bin/brew shellenv)"

**Comment by Homer712** [comment]
That doesn't look right. Let's try this again:
# Set PATH, MANPATH, etc., for Homebrew.
eval "$(/opt/homebrew/bin/brew shellenv)"

**Comment by Homer712** [comment]
Don't understand why the first line is bigger font and BOLD, but I tried it twice now, and it came out the same both times.

**Reply by gromgit** [reply]
> Don't understand why the first line is bigger font and BOLD, but I tried it twice now, and it came out the same both times.

GitHub comments use Markdown as a markup language. You actually [created a header](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#headings), when what you should've done is [quote your code](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code):
```
# Set PATH, MANPATH, etc., for Homebrew.
eval "$(/opt/homebrew/bin/brew shellenv)"
```

**Reply by p-linnane** [reply]
Delete both of those lines, save `.zprofile`, and relaunch your terminal to resolve your error.

**Reply by Homer712** [reply]
Thank You, Thank You, Thank You!

**Comment by Homer712** [comment]
Simple curiosity. This issue existed on my M1 MacBook Pro. The "zprofile" file existed, it's now basically an empty file, and everything is working perfectly. I could not find a "zshrc" file, but seeing how things are working, I have no worries.

On my Intel, 2018 Mac Mini (in which the Terminal has been working perfectly since the uninstall of Homebrew) I could not find either a "zprofile" file, or a "zshrc" file. I suspect that it the simply the difference between an M1 system, and an Intel system. But like I said, I'm curious and wondering if somebody would like to confirm that that is in fact the case.

**Reply by gromgit** [reply]
It's a simple issue of your login shell. Before macOS Catalina (introduced late 2019), users' default login shell would've been `bash`, not `zsh`. For that box, I would've asked you to look in `~/.bash_profile` or `~/.profile` instead. (More likely, I would've asked you for the output of `grep -l shelenv ~/.??*`, then asked you to edit some of the files listed.)

**Comment by Homer712** [comment]
I have now reinstalled Homebrew via the ".pkg" for MacOS. At the end of the install a window comes up with "Next Steps", which states:
To add Homebrew to your PATH, add to your shell profile (e.g. ~/.bash_profile or ~/.zprofile). On my M1 MacBook Pro I have found the ".zprofile" file and added the required text. But, on my 2018 Intel Mac Mini, I can not find either a ".zprofile" or a ".bash_profile". Everything seems to be running correctly with Homebrew and Terminal (I was even able to update my installation of Rsync), just curious as to where to find the either ".zprofile" or ".bash_profile" file on the Mac Mini. Both systems are running Sonoma 14.7.2.

**Reply by gromgit** [reply]
> But, on my 2018 Intel Mac Mini, I can not find either a ".zprofile" or a ".bash_profile".

As I stated previously, as long as you created your login under a version of macOS before Catalina, your login shell would be bash, not zsh. You can `echo $SHELL` to see which shell is active in your terminal session.

For bash, `open -a TextEdit ~/.bash_profile`, add the line shown in ***Next Steps***, then save the file and open a new terminal session. From that session, run `brew doctor` to make sure everything's OK. It shouldn't matter if the file didn't exist previously, as long as you edit the correct file for your shell.

**Reply by Homer712** [reply]
Thank you for continuing to help me through as I try to understand how things work with Homebrew. The M1 MacBook Pro was wiped clean due to a corrupt MS Office installation, and the MacOS installed was Monterey. The Intel 2018 Mac Mini I just got this year, and its first MacOS installed was also Monterey. Both systems are now on Sonoma, so both shells would be zsh.

If I understand "It shouldn't matter if the file didn't exist previously, as long as you edit the correct file for your shell." correctly, you're saying I can create the .zprofile file on the Mac Mini (or just copy it from the MacBook Pro) and install it in the same hidden location Users/My Name. I also discovered that the only text needed in the .zprofile file is "eval "$(/opt/homebrew/bin/brew shellenv)".

**Reply by gromgit** [reply]
> If I understand "It shouldn't matter if the file didn't exist previously, as long as you edit the correct file for your shell." correctly, you're saying I can create the .zprofile file on the Mac Mini (or just copy it from the MacBook Pro) and install it in the same hidden location Users/My Name. I also discovered that the only text needed in the .zprofile file is "eval "$(/opt/homebrew/bin/brew shellenv)".

Yes, and yes.

**Reply by Homer712** [reply]
Copied the hidden .zprofile file from the MacBook Pro, changed the "Next Steps" text to the Intel version and installed it on the Intel Mac Mini. Everything seems to be working perfectly. Thanks for your help and for allowing me to ask what must seem really basic questions.

**Reply by gromgit** [reply]
Oh, one thing I forgot to mention: the path to the `brew` binary changes between platforms:-
* Intel macOS: `/usr/local/bin/brew`
* ARM macOS: `/opt/homebrew/bin/brew`
* Linux: `/home/linuxbrew/.linuxbrew/bin/brew`

so you'll have to adjust the file(s) you copied over as appropriate.

**Reply by Homer712** [reply]
Sorry, but I'm confused again. I have two ".zprofile" files, one on the MacBook Pro, the other on the Mac Mini, both at location Users/My Name.

On the M1 Macbook Pro that file contains the text: eval "$(/opt/homebrew/bin/brew shellenv)"
On the Intel Mac Mini that file contains the text: eval "$(/usr/local/bin/brew shellenv)"

I think (if I've understood things correctly) that that is the required setup.

**Reply by gromgit** [reply]
Yes, that's as it should be.
